drt_1d.py

```diff
@@ -145,7 +241,7 @@ def read_input(file_path:str):
         raise e
     return data
 
-def perform_calculation(input_data:dict):
+def perform_calculation(input_data:dict) -> np.ndarray:
```

Comment:
I think you can pass in the components of the input data that you need so that this method doesn't need to depend on the structure of `input_data` but instead depends on clearly defined parameters that you can document in the docstring